docs/docs/vm-fundamentals/intro.md

```diff
@@ -2,6 +2,86 @@
 sidebar_position: -1
 ---
 
-# Intro
+# Introduction to the Cairo Virtual Machine
 
-An introduction to fundamentals. Which core concepts we will be seeing, what are their purpose etc. Maybe a little description of each one.
+Welcome to the documentation for the Cairo VM. 
+
+First, let us clarify what is a Virtual Machine (VM)? A VM is **a digital version of a physical computer,** that can run programs and operating systems, store data, connect to networks, and do other computing functions.
+
+This documentation is designed to provide developers with a fundamental understanding of the Cairo VM's architecture and capabilities, empowering them to build and optimize applications on Starknet effectively. This documentation will be divided into four main parts, each of which we will briefly describe in this introduction.
+
+The Cairo Virtual Machine (CVM), at the heart of the Starknet network, is an advanced computational framework specifically designed to execute Casm bytecode generated by the Cairo compiler. Also, the Cairo VM is able to generate cryptographic proofs, these proofs are be posted on a settlement layer such as Ethereum, offering Starknet the same level of security as the underlying layer
+
+## VM Layout
+
+The Cairo Virtual Machine optimizes its operations around a unique structure of registers and memory to facilitate the efficient execution of Casm bytecode. 
+
+### Registers
+
+Unlike traditional architectures where registers are separate storage units, in Cairo, these registers are pointers that facilitate access and management of memory during program execution.
+
+General-Purpose Registers: These are hypothetical in Cairo, where instead of traditional CPU-bound registers all operations are directly performed on memory cells. This approach aligns with Cairo’s optimization strategy which aims to minimize the number of trace cells in the Algebraic Intermediate Representation (AIR). In Cairo, registers such as the Program Counter (pc), Allocation Pointer (ap), and Frame Pointer (fp) do not store data themselves, instead:
+
+### Pointers
+
+Program Counter (pc) points to the current instruction in memory that the VM is executing.
+Allocation Pointer (ap) identifies the first free memory cell not yet used by the program, crucial for dynamic memory management. Frame Pointer (fp) points to the beginning of the stack frame of the current function, managing the execution context of function calls.
+
+### Memory
+
+Cairo employs a nondeterministic read-only memory model. Unlike typical random-access memory (RAM) used in most computing systems where any memory cell can be read or written at any time, Cairo's memory model only allows memory to be written once and restricts subsequent modifications. In addition, this model is particularly efficient for AIR, as it requires fewer trace cells per memory access, optimizing the proof generation process.
```

Comment:
i'm not sure `particularly efficient for AIR` is the proper way to say this.

Maybe  : `In addition, this memory model is particularly efficient for proof generation, as it requires fewer trace cells per memory access.`